src/main/java/io/pinecone/configs/PineconeConnection.java

```diff
@@ -145,14 +149,27 @@ private void onConnectivityStateChanged() {
                 channel.getState(false), channel);
     }
 
-    public static ManagedChannel buildChannel(String host) {
+    private ManagedChannel buildChannel(String host) {
```

Comment:
buildChannel is used within this class only so made it private. The method need not be static and the reason I had to remove the static keyword is because I wanted to use the PineconeConfig config which is private final. Also the reason why PineconeConfig can't be static is because not all connections will share the same config object.